src/Microsoft.Azure.Relay/Common/InputQueue.cs

```diff
@@ -638,7 +620,7 @@ class AsyncQueueWaiter : TaskCompletionSource<bool>, IQueueWaiter
             public AsyncQueueWaiter(CancellationToken cancellationToken, object state)
                 : base(state)
             {
-                this.cancelRegistration = cancellationToken.Register(s => CancelCallback(s), this);
+                this.cancelRegistration = cancellationToken.Register(CancelCallback, this);
```

Comment:
Undo this change for fewer allocations as mentioned above. (Yes I know VS2017 or some plugins try to correct this but I want fewer allocations, not slightly less code).